File: Timofeev-PE-42/Bizzare Underground C++/Project1/Key.cpp
#include "Key.h"
#include <iostream>

Key::Key() {
    std::cout << "--- Key created on the map ---" << std::endl;
}

Key::~Key() {
    std::cout << "--- Key destroyed. ---" << std::endl;
}

bool Key::characterKeyContact(int characterPositionX, int characterPositionY) {
    std::cout << "Key::characterKeyContact() called. Character at (" << characterPositionX << ", " << characterPositionY << ") found key." << std::endl;
    return true; // Предполагаем контакт для демонстрации
}

int Key::keyDestroy(int keyCounter) {
    std::cout << "Key::keyDestroy() called. Current key count: " << keyCounter << std::endl;
    int newCounter = keyCounter + 1;
    std::cout << "Key::keyDestroy(): Key picked up. New total keys: " << newCounter << std::endl;
    return newCounter;
}

File: Timofeev-PE-42/Bizzare-Underground-Java/Key.java
public class Key { // Не наследуется от GameObject в этом примере
    public Key() {
        System.out.println("Key created on the map.");
    }

    @Override
    protected void finalize() throws Throwable {
        System.out.println("Key destroyed.");
        super.finalize();
    }

    public boolean characterKeyContact(int characterPositionX, int characterPositionY) {
        System.out.println("Key.characterKeyContact() called. Character at (" + characterPositionX + ", " + characterPositionY + ") found key.");
        return true; // Предполагаем контакт
    }

    public int keyDestroy(int keyCounter) {
        System.out.println("Key.keyDestroy() called. Current key count: " + keyCounter);
        int newCounter = keyCounter + 1;
        System.out.println("Key.keyDestroy(): Key picked up. New total keys: " + newCounter);
        return newCounter;
    }
}